src/Contract/Frequency.php

```diff
@@ -32,4 +33,27 @@ public function getNextDue(DateTimeInterface $dateTime): DateTimeInterface;
      * @return bool
      */
     public function isExpired(DateTimeInterface $dateTime): bool;
+
+    /**
+     * Get the start time of this frequency
+     *
+     * @return ?DateTimeInterface
+     */
+    public function getStart(): ?DateTimeInterface;
+
+    /**
+     * Get the end time of this frequency
+     *
+     * @return ?DateTimeInterface
+     */
+    public function getEnd(): ?DateTimeInterface;
+
+    /**
+     * Parse the given json object and derive a {@see Frequency} instance from it
```

Comment:
`Create frequency from its JSON representation encoded with {@link json_encode()}`

Or even more precisely:

`Create frequency from its stored JSON representation previously encoded with {@link json_encode()}`

Which is close to [unserialize()](https://www.php.net/manual/en/function.unserialize.php).